core/src/main/java/org/apache/accumulo/fate/Fate.java

```diff
@@ -226,11 +223,17 @@ public Fate(T environment, TStore<T> store, Function<Repo<T>,String> toLogStrFun
    * Launches the specified number of worker threads.
    */
   public void startTransactionRunners(AccumuloConfiguration conf) {
-    int numThreads = conf.getCount(Property.MANAGER_FATE_THREADPOOL_SIZE);
-    executor = ThreadPools.createExecutorService(conf, Property.MANAGER_FATE_THREADPOOL_SIZE);
-    for (int i = 0; i < numThreads; i++) {
-      executor.execute(new TransactionRunner());
-    }
+    final ThreadPoolExecutor pool = (ThreadPoolExecutor) ThreadPools.createExecutorService(conf,
+        Property.MANAGER_FATE_THREADPOOL_SIZE);
+    fatePoolWatcher = ThreadPools.createGeneralScheduledExecutorService(conf);
+    fatePoolWatcher.schedule(() -> {
+      ThreadPools.makeResizeable(pool, conf, Property.MANAGER_FATE_THREADPOOL_SIZE);
+      int remaining = 1000 - pool.getQueue().size();
```

Comment:
Could replace 1000 w/ a function of the numThreads and time like below.

```suggestion
      // Assume a thread could execute up to 100 operations a second.  We are sleeping for three seconds. So to calculate max to queue do 3* 100 * numThreads. 
      int maxToQueue = 300 * conf.getCount(  Property.MANAGER_FATE_THREADPOOL_SIZE);
      int remaining = 1000 - pool.getQueue().size();
```

Reply:
Thanks for this, I wasn't sure what a reasonable value would be.

Reply:
I included this in 8bed9e5.